services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -6610,6 +6637,47 @@ public void killOfflinePush(String clusterName, String kafkaTopic, boolean isFor
     }
   }
 
+  /**
+   * If the child controllers are using the killJob topic as their current version, the killOfflinePush job should be aborted
+   */
+  public boolean isSafeToKillOfflinePush(String clusterName, String kafkaTopic, boolean isForcedKill) {
+    if (isForcedKill) {
+      return true;
+    }
+
+    String storeName = Version.parseStoreFromKafkaTopicName(kafkaTopic);
+    int versionNumber = Version.parseVersionFromKafkaTopicName(kafkaTopic);
+    Map<String, ControllerClient> controllerClients = getControllerClientMap(clusterName);
```

Comment:
Move this to parent helix controller and skill checking the current version in parent as parent controller store current version is not properly maintained.